### DISCUSSION TITLE
Ability to add a mini-page of information for a definition (that is independent of versions) in the Load/Import window

### DISCUSSION THREAD

**Discussion by nicolaasburgers** [nicolaasburgers/definition-library-releases#12] [Ideas]
This one page would be associated with a definition and would aim to provide general information about that definition's:
- purpose
- inputs and outputs
- limitations
- guidelines and general advice for use

This aims to fulfil the need to get a general gist about a definition that would be generally persistent and true throughout the life of definitions, across all versions.

This page could be displayed when a definition is selected in the search results, in a separate part of the window real estate to the current left or right results panels - perhaps the version history panel could be made to be smaller, to make room for a new panel containing this mini-page.

This could be saved in the repository as a markdown file with the same name (slug) of the definition so this can be rendered in the GitHub website, too.

**Comment by jakubczajczynski** [comment]
I like this idea a lot (because I had to ask for it too :)). Either separate window with te info page or at least a button that can open a readme in the webbrowser.

![image](https://github.com/user-attachments/assets/1f9e8db4-30a1-4c2a-8809-8b1500edd4dc)

**Comment by nicolaasburgers** [comment]
Actually this has largely already been implemented in 0.2.3 - so I need to update this thread!  If you right-click a cluster and enter text in the description field, it will be shown in the "More Info" window on the bottom left.  It seems like you have stumbled onto that too seeing as you have "jeje" in that window.

You can enter markdown in this description, too, and although currently that More Info window won't render it as markdown, it could be useful as it creates a .md page on GitHub that you can view, which does render it as markdown.

Your additional idea of the "Open Repository" button there is interesting. It could either be a button to load the description in the browser or the root of the repository - which one did you have in mind there?

**Reply by jakubczajczynski** [reply]
Nice, I like that it can be formatted as a Markdown. Would images and Mermaid graphs be possible?

I'm not sure about the "Open Repository" button, Maybe we should treat this as separate things.
1. Button to open repository - I'm leaning more towards opening the given GH definition localization rather than the root.
2. Opening a readme file of the definition. Even if you implement rendering it in the window, opening it in the web browser has its benefits. I must be able to pan, edit, save, etc. GH definitions while keeping the readme file opened, sometimes maybe more than one. There can't be any window focus conflicts, Keeping the readme opened and switching between windows while using Rhino.Inside.Revit can be problematic.

**Reply by nicolaasburgers** [reply]
The text you enter into the cluster's Description box can be anything - the More Info window will just display it as text.  The reason I mentioned markdown is because the contents are saved inside an .md file in the repo, making it render as markdown in a browser. 

So I guess you can put in anything that GitHub can render when you browse to an .md file stored in its repositories - which includes mermaid if I read pages like [this](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-diagrams).

And yes it makes sense that the button opening a link to the .md URL in a browser window could work fine as it does avoid window focus conflicts.  At the very least, implementing the button in this way is a very small piece of work and worth doing at least as a first step.

**Comment by mariomodest** [comment]
HI - first off I wanted to say that I love the definition library you created. This is a really powerful tool and solves a big problem that existed with GH definitions. Regarding the documentation for clusters, I was wondering if there is a way to add a hyperlink anywhere that is associated with the cluster. I was hoping it would be as simple as adding a link to the version comment or the cluster description but the link is not clickable and the text cannot be selected and copied (see below) - is there another way to do that?

![image](https://github.com/user-attachments/assets/cc6915ae-3bfd-4547-ac13-ea868cf1b6e3)

**Reply by nicolaasburgers** [reply]
I'm about to release a new version which offers a "view in browser" button next to that description box in the Import window which, when clicked, opens a browser tab at the address of the selected definition's documentation.
Would that (at least partly) answer your question?

**Reply by mariomodest** [reply]
So do you mean the selected definition's documentation on GitHub or would be any URL possible. So for example if I would want to have my cluster documentation in OneNote or a wiki page, etc.?

**Reply by nicolaasburgers** [reply]
Hi @mariomodest in one of the recent versions (0.3.x), any URLs appearing in that documentation box should now be clickable.  Hopefully that is what you had in mind!

**Reply by mariomodest** [reply]
@nicolaasburgers  yes that works perfect! Thanks for adding this feature
